collectors/python.d.plugin/nvidia_smi/nvidia_smi.chart.py

```diff
@@ -326,6 +332,22 @@ def name(self):
     def full_name(self):
         return 'gpu{0} {1}'.format(self.num, self.name())
 
+    @handle_attr_error
+    def pci_link_gen(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('pcie_gen').find('max_link_gen').text
+
+    @handle_attr_error
+    def pci_link_width(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('link_widths').find('max_link_width').text.split('x')[0]
+
+    def pci_bw_max(self):
+        link_gen = int(self.pci_link_gen())
+        link_width = int(self.pci_link_width())
+        # Maximum PCIe Bandwidth = SPEED * WIDTH * (1 - ENCODING) - 1Gb/s.
+        # see details https://enterprise-support.nvidia.com/s/article/understanding-pcie-configuration-for-maximum-performance
+        # return max bandwidth in kilobytes per second (kB/s)
+        return (PCI_SPEED[link_gen] * link_width * (1- PCI_ENCODING[link_gen]) - 1) * 1000 * 1000 / 8
```

Comment:
> PCI_SPEED = [2.5, 5, 8, 16, 32]
> PCI_ENCODING = [2/10, 2/10, 2/130, 2/130, 2/130]
> PCI_SPEED[link_gen]
> PCI_ENCODING[link_gen])

I think this is wrong

```xml
<pcie_gen>
	<max_link_gen>3</max_link_gen>
</pcie_gen>
```

> > where 2.5GT/s is referred as "gen1", 5GT/s as "gen2", and 8GT/s as "gen3"

PCI_SPEED[3] == 16, but should be 8.

You need to either use `link_gen - 1` as an index or make `PCI_SPEED` a dictionary (link_gen: speed).

```python
PCI_SPEED = {
  "1": 2.5, 
  "2": 5, 
  "3": 8, 
  "4": 16,
  "5": 32
}
```